drivers/adc/ad7606/ad7606.c

```diff
@@ -1495,8 +1495,11 @@ int32_t ad7606_set_config(struct ad7606_dev *dev,
 	if (dev->sw_mode) {
 
 		val |= no_os_field_prep(AD7606_CONFIG_OPERATION_MODE_MSK, config.op_mode);
-		if ((uint8_t)config.dout_format > (uint8_t)dev->max_dout_lines)
+		if ((uint8_t)config.dout_format > (uint8_t)dev->max_dout_lines) {
+			printf("Device misconfig: max DOUT is %u, but trying to set %u\n",
```

Comment:
If the project that uses this device driver has an running iiod that communicates with an iio client over serial, the comm will be messed up by these messages. It's pretty common that we redirect printf output to serial. I'd drop this one change.